pgx-pg-sys/src/submodules/oids.rs

```diff
@@ -9,41 +9,188 @@ Use of this source code is governed by the MIT license that can be found in the
 
 #![allow(non_camel_case_types)]
 use crate as pg_sys;
-use crate::PgBuiltInOids;
+use crate::BuiltinOid;
+use crate::Datum;
+use pgx_sql_entity_graph::metadata::{
+    ArgumentError, Returns, ReturnsError, SqlMapping, SqlTranslatable,
+};
 
-impl PgBuiltInOids {
-    pub fn value(self) -> pg_sys::Oid {
-        self as isize as pg_sys::Oid
+/// An [object identifier][pg_docs_oid] in Postgres.
+///
+/// This is meant to be understood purely by equality. There is no sensible "order" for Oids.
```

Comment:
There might not be a sensible order for Oids, but Postgres does indeed support `ORDER BY oid` clauses, and they're technically just u32s.

I can see situations where an extension might want to store a bunch of Oids in a BTreeSet or as keys in a BTreeMap.

This is the only thing I see in the PR that we might want to re-consider, so I'm not going to let that stop me from smashing the merge button once I get CI to pass it...

Reply:
If anyone objects I'm happy to see us add the derive, it's just usually adding that impl suggests the comparison is meaningful, where here it's... kinda not.

Reply:
That's fine.  No problem.